content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,122 @@
+
+This enables a smooth transition for end users. Additionally as storage plugin developers, we can reduce the burden of maintaining the in-tree storage plugins and eventually remove them from the core Kubernetes binary.
+
+## What has been changed and What's new?
+
+Since Kubernetes v1.17, there are some new changes introduced:
+
+### New Feature Gate
+
+`CSIMigration{provider}Complete` was introduced before as a supplementary feature gate once CSI migration is enabled on all of the nodes. This flag will de-register the in-tree storage plugin and directly translate the traffic to CSI driver without any checking if CSI migration is enabled on the node.
+
+While this feature gate is a great helper, we would also like a feature gate that can disable in-tree storage plugin without CSI migration enabled. For example, we would like to disable ebs storage plugin on a GKE cluster because it does not have ability to provision EBS volume. Thus, a new feature flag `InTreePlugin{vendor}Unregister` is introduced in kubernetes v1.21.
+
+`InTreePlugin{vendor}Unregister` is a standalone feature gate that can be enabled and disabled out of CSI Migration scope. The name speaks for itself, when enabled, the component will not register the specific in-tree storage plugin to the supported list. If the cluster operator only enables this flag, end users will get an error from PVC saying it cannot find the plugin when the plugin is used. The cluster operator may want to enable this regardless of CSI Migration if they do not want to support the legacy in-tree APIs and only support CSI moving forward.
+
+At the same time, `CSIMigration{provider}Complete` was deprecated and removed in kubernetes v1.21 as well. The `InTreePlugin{vendor}Unregister` will replace the old feature flag but remain all the functionality that `CSIMigration{provider}Complete` provides.
+
+### Observability
+
+We have added the [CSI Migration Metrics](https://github.com/kubernetes/kubernetes/issues/98279) support for better observability of the feature. This will also help us discover any potential issues regarding the feature.
+
+- storage_operation_duration_seconds
+  - This metric is exposed from the Kube-Controller-Manager. A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_sidecar_operations_seconds
+  - This metric is exposed from the CSI sidecars(csi-provisioner, csi-attacher, csi-resizer). A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_kubelet_operations_seconds
+  - This metric is exposed from Kubelet. It exposes CSI operation, driver name, status and `migrated` field to indicate whether this storage operation is a CSI migration operation.
```

Comment:
I'd write this as a table (in HTML), with a column for the component name, a column for the metric name, and a column for notes. Plus headings, `<tbody>`, etc.
In the notes, maybe list the common _labels_ (not fields) such as `migrated`. Either link to the reference docs for these metrics, or also explain the values for `migrated` (I presume: `0.0` or `1.0`, but it might be present / absent - I don't know!).

Reply:
Thanks for the suggestion! I changed it to a table.